sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -283,9 +296,15 @@ private Mono<Void> refreshLocation(boolean isReadRequest, boolean forceRefresh,
         return this.globalEndpointManager.refreshLocationAsync(null, forceRefresh);
     }
 
-    private Mono<ShouldRetryResult> shouldRetryOnBackendServiceUnavailableAsync(boolean isReadRequest, boolean isWebExceptionRetriable) {
-        if (!isReadRequest && !isWebExceptionRetriable) {
-            logger.warn("shouldRetryOnBackendServiceUnavailableAsync() Not retrying on write with non retriable exception. Retry count = {}", this.serviceUnavailableRetryCount);
+    private Mono<ShouldRetryResult> shouldRetryOnBackendServiceUnavailableAsync(
+        boolean isReadRequest,
+        boolean isWebExceptionRetriable,
+        boolean nonIdempotentWriteRetriesEnabled) {
+
+        if (!isReadRequest && !nonIdempotentWriteRetriesEnabled && !isWebExceptionRetriable) {
```

Comment:
for write operation, is server generated 503/x safe to retry?

Reply:
Yes, but only if retriable writes is enabled - so, above code should be fine